src/wp-includes/media.php

```diff
@@ -5627,6 +5627,25 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 
 	$loading_attrs = array();
 
+	/**
+	 * Filters whether to short-circuit loading optimization attributes.
+	 *
+	 * Returning an array from the filter will effectively short-circuit the loading of optimization attributes,
+	 * returning that value instead.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array  $loading_attrs The loading optimization attributes.
+	 * @param string $tag_name      The tag name.
+	 * @param array  $attr          Array of the attributes for the tag.
+	 * @param string $context       Context for the element for which the loading optimization attribute is requested.
+	 */
+	$loading_attrs = apply_filters( 'pre_wp_get_loading_optimization_attributes', $loading_attrs, $tag_name, $attr, $context );
+
+	if ( ! empty( $loading_attrs ) ) {
```

Comment:
See https://github.com/WordPress/wordpress-develop/pull/5176#discussion_r1330580812, the filtered type was initially correct and IMO should be reverted to that.
```suggestion
	 * @param array|false  $loading_attrs False by default, or array of loading optimization attributes to short-circuit.
	 * @param string       $tag_name      The tag name.
	 * @param array        $attr          Array of the attributes for the tag.
	 * @param string       $context       Context for the element for which the loading optimization attribute is requested.
	 */
	$loading_attrs = apply_filters( 'pre_wp_get_loading_optimization_attributes', false, $tag_name, $attr, $context );

	if ( is_array( $loading_attrs ) ) {
```

Reply:
Thanks, @felixarntz for the clarification here. I was pretty sure I'd seen this pattern before.